engine/robot_state.py

```diff
@@ -115,60 +115,65 @@ def __init__(self, **kwargs):
+        self.width_margin = kwargs.get("width_margin", 1)
+        self.threshold_distance = kwargs.get("threshold_distance", ((
+            self.width + self.width_margin) / 2) / math.cos(math.radians((180 - self.sensor_measuring_angle) / 2)))
+        self.detect_obstacle_range = kwargs.get("detect_obstacle_range", min(
+            self.threshold_distance, self.max_sensor_range))  # set ultrasonic detection range
 
         # MEASUREMENTS
-        self.state = kwargs.get("state", np.array([[kwargs.get("xpos", 0)], [kwargs.get("ypos", 0)], [kwargs.get("heading", 0)]]))
-        self.truthpose = kwargs.get("truthpose", np.transpose(np.array([[kwargs.get("xpos", 0)], [kwargs.get("ypos", 0)], [kwargs.get("heading", 0)]])))
+        self.state = kwargs.get("state", np.array(
+            [[kwargs.get("xpos", 0)], [kwargs.get("ypos", 0)], [kwargs.get("heading", 0)]]))
+        self.truthpose = kwargs.get("truthpose", np.transpose(np.array(
+            [[kwargs.get("xpos", 0)], [kwargs.get("ypos", 0)], [kwargs.get("heading", 0)]])))
         self.plastic_level = kwargs.get("plastic_level", 0)
         self.battery = kwargs.get("battery", 100)
         self.acceleration = kwargs.get("acceleration", [0, 0, 0])
         self.magnetic_field = kwargs.get("magnetic_field", [0, 0, 0])
         self.gyro_rotation = kwargs.get("gyro_rotation", [0, 0, 0])
         self.init_gps = kwargs.get("init_gps", (0, 0))
-        self.gps_data = kwargs.get("gps_data",(0, 0) )
-        self.imu_data = kwargs.get("imu_data", None)  # will be filled by execute_setup
+        self.gps_data = kwargs.get("gps_data", (0, 0))
+        # will be filled by execute_setup
+        self.imu_data = kwargs.get("imu_data", None)
         self.linear_v = kwargs.get("linear_v", 0)
         self.angular_v = kwargs.get("angular_v", 0)
-        self.dist_to_goal = kwargs.get("dist_to_goal", 0)
         self.prev_phase = kwargs.get("prev_phase", self.phase)
         self.goal_location = kwargs.get("goal_location", (0, 0))
-       
+
         # SENSORS
         self.motor_controller = kwargs.get("motor_controller", None)
         self.robot_radio_session = kwargs.get("robot_radio_session", None)
         self.gps = kwargs.get("gps", None)
         self.imu = kwargs.get("imu", None)
         self.ekf = kwargs.get("ekf", None)
-        self.front_ultrasonic = kwargs.get("front_ultrasonic", None)
-        self.lf_ultrasonic = kwargs.get("lf_ultrasonic",None )
-        self.lb_ultrasonic = kwargs.get("lb_ultrasonic", None)
-        self.rf_ultrasonic = kwargs.get("rf_ultrasonic", None)
-        self.rb_ultrasonic = kwargs.get("rb_ultrasonic", None)
 
         # TODO: GPS, IMU, RF module and Motor Controller are also re-initialized in robot.execute_setup
         # We should pick whether we want to initialize the attributes here or in robot.execute_setup
         # (which is being passed from Mission, need to whether sensors are part of mission vs robot)
```

Comment:
https://github.com/cornellnexus/src/pull/122#discussion_r1148410593

Reply:
this isn't part of my change